### ISSUE TITLE
Potential MPI issue when using GPU kernel of ELPA on a subset of MPI ranks

### ISSUE THREAD

**Issue by dsambit** [marekandreas/elpa#1]
Dear ELPA developers,

We have been successfully using CPU kernels of ELPA in DFT-FE code (https://github.com/dftfeDevelopers/dftfe), where I am one of the lead developers. We use ELPA on a subset of MPI ranks of the MPI_COMM_WORLD used by DFT-FE. In particular, we use MPI_Comm_create_group to create the new communicator which we pass to ELPA.  Recently, I have been trying to use the GPU kernels of ELPA (using elpa-2020.11.001.rc1) on Summit supercomputer using the same route, but the code gets stuck after printing:

`Initializing the GPU devices
`
when using subset of the ranks. It works fine if all ranks are used.

Upon investigating the ELPA source code, I found the issue to be the call to mpi_allreduce where MPI_COMM_WORLD is used. 
https://github.com/marekandreas/elpa/blob/5bff935698a6826321d659666e690574b78777c1/src/GPU/check_for_gpu.F90#L101

```
#ifdef WITH_MPI
      call mpi_allreduce(numberOfDevices, maxNumberOfDevices, 1, MPI_INTEGER, MPI_MAX, MPI_COMM_WORLD, mpierr)

      if (maxNumberOfDevices .ne. numberOfDevices) then
        print *,"Different number of GPU devices on MPI tasks!"
        print *,"GPUs will NOT be used!"
        gpuAvailable = .false.
        return
      endif
#endif
```

Since this is a sanity check, I have temporarily bypassed the above issue by commenting the above lines of code. Now I am able to run the GPU-kernel successfully on subset of ranks. 

I would be very grateful for any guidance in resolving this issue cleanly either in the way we interfacing with ELPA, or if it could be fixed in the ELPA source code itself.

Thank you,
Sambit

**Comment by marekandreas** [comment]
This issue is fixed in commit 11a6c6d8

**Comment by dsambit** [comment]
Thank you for making the fix.
